File: app/javascript/components/HistorySection/History.js
import React from "react";

function History({ changeType, oldVal, newVal, changeDate }) {
  return (
    <tr>
      <td className="fs-4">{changeType}</td>
      <td className="fs-4">{oldVal}</td>
      <td className="fs-4">{newVal}</td>
      <td className="fs-4">{changeDate}</td>
    </tr>
  );
}

export default History;

File: app/models/history.rb
class History < ApplicationRecord
  belongs_to :changed_by, class_name: "User", inverse_of: :histories
  belongs_to :ticket

  CHANGE_TYPES = ["assigned dev", "priority", "title", "description", "submitted by", "status", "ticket type" ]
  validates :changed_by, :ticket, :old_value, :new_value, :change_type, presence: true
  validates :change_type, inclusion: { in: CHANGE_TYPES, message: "Change type is not valid" }
end
